beeline/propagation/honeycomb.py

```diff
@@ -43,19 +44,31 @@ def marshal_propagation_context(propagation_context):
     # FIXME: Since ALL trace fields are in propagation_context, we may want to strip
     # some automatically added trace fields that we DON'T want to propagate - e.g. request.*
     version = 1
+    dataset = None
+    if propagation_context.dataset:
+        dataset = quote(propagation_context.dataset)
     trace_fields = base64.b64encode(json.dumps(
         propagation_context.trace_fields).encode()).decode()
-    trace_header = "{};trace_id={},parent_id={},context={}".format(
-        version, propagation_context.trace_id, propagation_context.parent_id, trace_fields
+    trace_header = "{};dataset={},trace_id={},parent_id={},context={}".format(
```

Comment:
If dataset is `None`, we would end up with the header value of `1;dataset=,trace_id=...` - is it okay to have an empty dataset value?

Reply:
I'll see how the other beelines handle this. If you don't have a dataset then the beeline won't be sending any events

Reply:
looks like it's pretty variable across them so I'll add some more defensiveness here